beancount_import/matching.py

```diff
@@ -255,6 +283,81 @@ def add_posting(self, entry: Transaction, mp: MatchablePosting):
         group = self._postings.setdefault(_date_key(entry, mp), {})
         group[source_posting_ids] = (entry, mp)
 
+        weight = get_posting_weight(mp.posting)
+        if weight is None:
+            return
+
+        for dc in self.fuzz_date_currency_key(self._date_currency_key(entry, mp)):
+            bisect.insort(
```

Comment:
It would be better to lazily-sort these lists, i.e. add to the end when inserting, and record that the list is not sorted.  Then when querying, sort the list if it isn't already sorted.

When using `insort` as you have here, adding `n` postings with the same currency and date is `O(n**2)`.